### PR TITLE
Japheth portfoliocalculator

### PR DESCRIPTION
**Pull request by japhethleongyh**
added pnl and dpnl to the individual stocks in portfoliocalculator

### CODE REVIEW

File: backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculator.java [hunk 1]
```diff
@@ -22,5 +22,7 @@ public class PortfolioCalculator {
     private double avgStockBuyPrice;
```
Comment (quinncheong) [inline backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculator.java:22]: This file needs refactoring. Portfolio Calculator should be static, and uses DI to take in stocks/portfolios.

A calculator does not have a daily PNL.

File: backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculatorService.java [hunk 2]
```diff
@@ -15,7 +15,10 @@
 import com.is442g1t4.gpa.portfolio.allocatedStock.AllocatedStockService;
 import com.is442g1t4.gpa.portfolio.allocatedStock.AllocatedStockRepository;
 
+import java.time.LocalDate;
```
Comment (quinncheong) [inline backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculatorService.java:18]: Do we need to use LocalDate, ZoneId, Instant? Is there a way to make it less complicate and remove the use of these?

File: backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculatorService.java [hunk 3]
```diff
@@ -33,24 +36,42 @@ public Map<String, PortfolioCalculator> getCalculatedStockInPortfolio(ObjectId i
 
             for (AllocatedStock allocatedStock : allocatedStocks){
```
Comment (quinncheong) [inline backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculatorService.java]: Lines 37 - 70+ is too long. Refactor this component.

**Review by quinncheong — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculator.java (hunk 1 above)
Comment (japhethleongyh) [reply backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculator.java:22]: I tried making it static initially but when you make it static it conflicts with springboot's autowiring

File: backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculatorService.java (hunk 2 above)
Comment (japhethleongyh) [reply backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculatorService.java:18]: This implementation makes it so the code is region agnostic and doesn't depend on the datetime in the db being the same as local/machine datetime

File: backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculator.java (hunk 1 above)
Comment (japhethleongyh) [reply backend/src/main/java/com/is442g1t4/gpa/portfolio/portfolioCalculator/PortfolioCalculator.java:22]: Also the calculator is misnamed now, this is how Elijah named it previously but what this does is to calculate the stats of each stock inside a portfolio, so each portfoliocalculator object is actually one calculated and aggregated stock in a portfolio

**Pull request closed without merging**